Sources/Sharing/SharedPublisher.swift

```diff
@@ -15,9 +15,9 @@
     /// }
     /// ```
     public var publisher: some Publisher<Value, Never> {
-      box.subject
-        .handleEvents(receiveSubscription: { [box] _ in _ = box })
-        .prepend(wrappedValue)
+      Just<Void>(()).flatMap { _ in
+        box.subject.prepend(wrappedValue)
+      }
```

Comment:
AFK now, but I wonder if you could replace this whole thing with just:
```suggestion
     box.reference.publisher
```
?

Reply:
Also AFK, but I don't think so because the subjects these references have are passthrough subjects, which is why prepending the current value is needed when creating a new publisher.

Reply:
If you look at the available implementations for `Reference`-conforming types, you'll find that they already prepend the current value, e.g. https://github.com/pointfreeco/swift-sharing/blob/main/Sources/Sharing/Internal/Reference.swift#L58-L60

Reply:
@pyrtsa That prepending is there for when you reload a shared with a new projected value and is unrelated to these properties directly on `Shared` (if you try locally you can see which tests will fail).